packages/http/src/metadata.ts

```diff
@@ -580,7 +592,7 @@ export function createMetadataInfo(program: Program, options?: MetadataInfoOptio
       // For OpenAPI emit, for example, this means that we won't put a
       // readOnly: true property into a specialized schema for a non-read
       // visibility.
-      keepShareableProperties ||= visibility === canonicalVisibility;
+      keepShareableProperties ??= visibility === canonicalVisibility;
```

Comment:
Split this internal param into another function and change to be more explicit about what this flag means